apps/studio/src/pageComponents/main/JSCodeshiftRender.tsx

```diff
@@ -29,7 +31,15 @@ const MonacoDiffEditor = dynamic(
 
 export const useCodeDiff = () => {
 	const { setEvents, events } = useLogStore();
-	const { engine, inputSnippet, afterInputRanges } = useSnippetStore();
+	const { engine, afterInputRanges } = useSnippetStore();
+
+	const { selectFirst } = useFilesStore();
+
+	const inputSnippet =
+		selectFirst({
+			parent: DEFAULT_TEST_FIXTURE_DIR.hashDigest,
+			name: "before.tsx",
```

Comment:
the idea to treat snippets like files is very good.  
how about splitting extension to another property?